### ISSUE TITLE
Permit searching for other pilots who permit it

### ISSUE THREAD

**Issue by KayRJay** [ericberman/MyFlightbookWeb#444]
MyFlightBook has information on thousands of pilots.  Wouldn't it be nice to permit pilots to publish (subsets) of their information, and make that searchable?  It would be a great way to meet other pilots.

Obviously, users should be able to selectively choose to publish their names, home airport, email, phone, address, achievements, ratings, airplane, etc.   Then, there could be a search page for locating pilots by any/most of these items.

You could also support searching in other ways:

- Pilots who've visited a given airport (within in range of dates)
- Pilots who've flown a given route (within a range of dates)
- Pilots who own a given aircraft model
- (probably others ...)

**Comment by ericberman** [comment]
Yeah, I've actually had a similar feature request on my plate for years now - the idea that you can "friend" other pilots on the system and let them see various stuff.  Can already do that to some degree with student/instructor relationships, but would be a bit more general and not permit things like signing or endorsements.

Your suggestion here takes it in a more social direction, less about sharing data than about making connections.

**Comment by KayRJay** [comment]
Actually, both sharing and searching are useful and complementary.

**Comment by KayRJay** [comment]
And ...
- Pilots who have flown a specific aircraft tail number
- Pilots who have flown a specific aircraft type/category
- Pilots with a specific "home" airport

And ... a user could list in his profile his aviation-related clubs and usernames at various online sites: AOPA, Cirruspilot, etc.  

And ... a search for users/pilots would result in a list (much like searching for airports).  Clicking on a user name would display that part of a profile that the user wishes to make public.

**Comment by ericberman** [comment]
I think all of this is very interesting.  No idea if/when I'll get to it, but both dimensions (sharing data with people you already know, and finding people you may not know) I think have potential.

**Comment by KayRJay** [comment]
Thanks for considering this.

**Comment by KayRJay** [comment]
By the way, the queries like (some of) the above (and optimizing them) might relate to the queries about airports discussed in issue #448, but I'm sure you already realized that!

**Comment by KayRJay** [comment]
Queries about published flights by a specific pilot other than the current user:

- N most recent flights
- flights within a date range
- flights in a specific aircraft type
- flights in a specific aircraft tail number
- flights visiting a specific (list of) airport(s)

You could also support similar queries about flights by all members of a specific flying club (maybe only a club I belong to or manage).

**Comment by ericberman** [comment]
BTW, just realized that I should have updated this issue about 6 months ago when I added the capability to share your logbook with others in a read-only manner.  Profile->Preferences->Sharing - you can say what to share and create a specific link for a specific person (or group of people) and revoke it if you like.

Doesn't cover any of the searching functionality you suggest, but it does cover the "friending" piece described.

**Comment by KayRJay** [comment]
Thanks!    Hope you’ll be able to get to the searching ideas sometime too ...
